### ISSUE TITLE
implementations as attribute into the instance

### ISSUE THREAD

**Issue by loic001** [Scille/umongo#341]
First of all, congratulations for this work!

As mentioned [here](https://umongo.readthedocs.io/en/latest/_modules/umongo/instance.html?highlight=auto%20register#) "implementations are registered as attribute into the instance".

But it doesn't seem to work... It seems that the `retrieve_document' function is made for this but I would have preferred an instance.MyImplementedDocument approach.

Can you shed some light on this?

Note that I don't want to use register as a decorator, to decouple the documents from the backend choice.

Thanks for your help!

**Comment by lafrech** [comment]
Indeed, looks like the docstring is outdated. The note below doesn't mention Mixin.

I think I removed this because

- I didn't see a use case
- There could be name clashes if a Doc and an Embedded doc have the same name that could catch the user by surprise.

Do you really have a use case for this?

If you use `instance.Doc` somewhere, why not just import `Doc`?
